test/unit/cloudantBaseService.test.js

```diff
@@ -152,24 +157,60 @@ describe('Test CloudantBaseService', () => {
     );
   });
 
-  it('Use basic authenticator', () => {
+  it('Use IAM authenticator', () => {
     const auth = new IamAuthenticator({
       apikey: '1234',
       clientId: '345',
       clientSecret: '234',
     });
     const service = new CloudantBaseService({ authenticator: auth });
     assert.ok(service);
-    // no new jar was generated
-    assert.ok(service.baseOptions.jar === undefined);
+    // cookie jar exists
+    assert.ok(service.baseOptions.jar);
+    assert.strictEqual(
+      service.baseOptions.jar,
+      service.requestWrapperInstance.axiosInstance.defaults.jar
+    );
     service.setServiceUrl(newUrl);
     assert.strictEqual(service.getAuthenticator().tokenManager.url, iamUrl);
     service.configureService('apple');
-    // no new jar and url were set
-    assert.ok(service.baseOptions.jar === undefined);
+    // cookie jar exists
+    assert.ok(service.baseOptions.jar);
+    assert.strictEqual(
+      service.baseOptions.jar,
+      service.requestWrapperInstance.axiosInstance.defaults.jar
+    );
     assert.strictEqual(service.getAuthenticator().tokenManager.url, iamUrl);
   });
 
+  it('Custom cookie jar with basic authenticator', () => {
+    const auth = new BasicAuthenticator({
+      username: 'test',
+      password: 'user', // pragma: allowlist secret
+    });
+    const testJar = new CookieJar(new MemoryCookieStore(), {
+      rejectPublicSuffixes: false,
+    });
+    const service = new CloudantBaseService({
+      authenticator: auth,
+      jar: testJar,
+    });
+    assert.ok(service);
+    // cookie jar exists
+    assert.ok(service.baseOptions.jar);
+    // cookie jar is the same as expected test jar
+    let axiosJar = service.requestWrapperInstance.axiosInstance.defaults.jar;
+    assert.strictEqual(axiosJar, testJar);
+    assert.strictEqual(axiosJar.rejectPublicSuffixes, false);
+    service.setServiceUrl(newUrl);
+    service.configureService('apple');
+    // cookie jar is the same as expected test jar
+    assert.ok(service.baseOptions.jar);
```

Comment:
I'd bring up this comment before the resets and separate it from the previous assertions in space.

```suggestion
    
    // reconfigure the service
    service.setServiceUrl(newUrl);
    service.configureService('apple');
    // cookie jar still exists
    assert.ok(service.baseOptions.jar);
    // cookie jar is still the same as expected test jar
```